app/models/section_offering.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+# == Schema Information
+#
+# Table name: section_offerings
+#
+#  id         :bigint           not null, primary key
+#  title      :text(65535)      not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+
+class SectionOffering < ApplicationRecord
+  has_and_belongs_to_many :sections,
+    class_name: Group::Sektion.sti_name,
+    association_foreign_key: :group_id
```

Comment:
Not a hudge fan of the habtm relation (explicit join models help), but it was in the requirements like this.